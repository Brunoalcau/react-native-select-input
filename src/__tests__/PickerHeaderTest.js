import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';
import { shallow } from 'enzyme';
import PickerHeader from '../PickerHeader';

describe('PickerHeader', () => {
  beforeEach(() => {
    this.props = {
      done: 'Done',
      toggleShowOptions: jest.fn(),
    };
  });

  it('renders correctly', () => {
    const wrapper = shallow(<PickerHeader {...this.props} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should close options if pressed', () => {
    const wrapper = shallow(
      <PickerHeader
        {...this.props}
        visible
      />,
    );

    wrapper.find(TouchableWithoutFeedback).props().onPress();

    expect(this.props.toggleShowOptions).toHaveBeenCalled();
  });
});
