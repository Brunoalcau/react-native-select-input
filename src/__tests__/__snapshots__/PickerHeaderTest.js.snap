// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#F8F8F8",
      "paddingHorizontal": 25,
      "paddingVertical": 10,
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 14,
            },
            null,
          ]
        }
      >
        Done
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`PickerHeader renders correctly with a testPropterty set 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#F8F8F8",
      "paddingHorizontal": 25,
      "paddingVertical": 10,
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
    testID="header-foo"
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 14,
            },
            null,
          ]
        }
      >
        Done
      </Text>
    </View>
  </TouchableWithoutFeedback>
</View>
`;
